tensorflow-framework/src/main/java/org/tensorflow/framework/metrics/impl/ConfusionMatrixConditionCount.java

```diff
@@ -0,0 +1,196 @@
+   *     prediction values to determine the truth value of predictions (i.e., above the threshold is
+   *     {@code true}, below is {@code false}). One metric value is generated for each
+   *     threshold value.
+   * @param seed the seed for random number generation. An initializer created with a given seed
+   *     will always produce the same random tensor for a given shape and data type.
+   * @param type the data type for the variables
+   */
+  public ConfusionMatrixConditionCount(
+      Ops tf,
+      String name,
+      ConfusionMatrixEnum confusionMatrixCond,
+      float[] thresholds,
+      long seed,
+      Class<T> type) {
+    super(tf, name, seed);
+    accumulatorName = this.getVariableName(ACCUMULATOR);
+    this.type = type;
+    this.confusionMatrixCond = confusionMatrixCond;
+    this.thresholds = thresholds;
+    init();
+  }
+
+  /** Initialize the metric */
+  private void init() {
+    Shape variableShape = Shape.of(this.thresholds.length);
+
+    Zeros<T> zeros = new Zeros<>(getTF());
+    accumulator =
+        getTF()
+            .withName(getAccumulatorName())
+            .variable(zeros.call(getTF().constant(variableShape), type));
+    initializer = getTF().assign(accumulator, zeros.call(getTF().constant(variableShape), type));
+  }
+
+  /**
+   * Gets the initializer for the accumulator variable
+   *
+   * @return the initializer for the accumulator variable
+   */
+  public Assign<T> getInitializer() {
+    return initializer;
+  }
+
+  /**
+   * Accumulates the metric statistics.
+   *
+   * @param labels The ground truth values.
+   * @param predictions the predictions
+   * @param sampleWeights Optional weighting of each example. Defaults to 1. Rank is either 0, or
+   *     the same rank as labels, and must be broadcastable to labels.
+   * @return a List of Operations to update the metric state.
+   */
+  @Override
+  public List<Op> updateStateList(
+      Operand<? extends TNumber> labels,
+      Operand<? extends TNumber> predictions,
+      Operand<? extends TNumber> sampleWeights) {
+    Ops tf = getTF();
+    Operand<T> tLabels = cast(tf, labels, type);
+    Operand<T> tPredictions = cast(tf, predictions, type);
+    Operand<T> tSampleWeights = sampleWeights != null ? cast(tf, sampleWeights, type) : null;
+    return new ArrayList<>(
+        MetricsHelper.updateConfusionMatrixVariables(
+            getTF(),
+            Collections.singletonMap(confusionMatrixCond, accumulator),
+            Collections.singletonMap(confusionMatrixCond, initializer),
+            tLabels,
+            tPredictions,
+            tf.constant(thresholds),
+            null,
+            null,
+            tSampleWeights,
+            false,
+            null));
+  }
+
+  /** {@inheritDoc} */
+  @Override
+  public Operand<T> result() {
+    return getTF().identity(accumulator);
+  }
+
+  /** {@inheritDoc} */
+  @Override
+  public Op resetStates() {
```

Comment:
The `initializer` is an `Assign` Operation, and when run, will assign zeroes to the `accumulator`, which is the stored value.
So each time `resetStates` is run, the `accumulator` will be reset to zeroes. 

This Op could be used with a control dependency on `accumulator`, but I elected to let the caller do that if wanted.  The issue is that this Op should not be permanently attached to `accumulator` because if you are doing, for example, multiple training steps you do not want to always reset it to zero on each step, but rather add to the it. However, the caller may want to reset the value to zeros, perhaps between epochs.

In the test cases I call
`session.run(instance.getInitializer());`
which could have equally been
`session.run(instance.resetStates());`